lang/py/avro/schema.py

```diff
@@ -136,7 +147,18 @@ def _is_timezone_aware_datetime(dt):
 # Base Classes
 #
 
-class Schema(abc.ABC):
+class CanonicalPropertiesMixin(object):
+    """A Mixin that provides canonical properties to Schema and Field types."""
+    @property
+    def canonical_properties(self):
+        props = self.props
+        return collections.OrderedDict(
```

Comment:
Should we just use a dictionary here?

Reply:
While keys are inserted in the order specified by the canonical form, dictionary order is guaranteed only in Python 3.7 and above. So I believe we still need this to be `OrderedDict`. 

If we were to stop supporting 3.6 someday, we could revert to using a dictionary; I believe it would work without any changes.